### ISSUE TITLE
Neuron not working with Github v4

### ISSUE THREAD

**Issue by davefp** [uesteibar/neuron#19]
I'm new to both GraphQL and Elixir, so I apologize if there's something obvious here that I'm missing.

I'm trying to use Neuron to talk to the Github v4 GraphQL API. Here's my code:
```
    url = "https://api.github.com/graphql"
    headers = [Authorization: "bearer <REDACTED>"]
    query = "{ viewer { login }}"
    Neuron.Config.set(url: url)
    Neuron.Config.set(headers: headers)
    resp = Neuron.query(query)
    IO.puts inspect resp
```
Here's the content of `resp`:
```
{:error, %Neuron.Response{body: %{"documentation_url" => "https://developer.github.com/v4", "message" => "Problems parsing JSON"}, headers: [{"Server", "GitHub.com"}, {"Date", "Wed, 22 Aug 2018 08:59:57 GMT"}, {"Content-Type", "application/json; charset=utf-8"}, {"Content-Length", "89"}, {"Status", "400 Bad Request"}, {"X-RateLimit-Limit", "5000"}, {"X-RateLimit-Remaining", "4999"}, {"X-RateLimit-Reset", "1534931646"}, {"Cache-Control", "no-cache"}, {"X-OAuth-Scopes", ""}, {"X-Accepted-OAuth-Scopes", "repo"}, {"X-OAuth-Client-Id", "REDACTED"}, {"X-GitHub-Media-Type", "github.v4; format=json"}, {"Access-Control-Expose-Headers", "ETag, Link, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval"}, {"Access-Control-Allow-Origin", "*"}, {"Strict-Transport-Security", "max-age=31536000; includeSubdomains; preload"}, {"X-Frame-Options", "deny"}, {"X-Content-Type-Options", "nosniff"}, {"X-XSS-Protection", "1; mode=block"}, {"Referrer-Policy", "origin-when-cross-origin, strict-origin-when-cross-origin"}, {"Content-Security-Policy", "default-src 'none'"}, {"X-Runtime-rack", "0.027861"}, {"X-GitHub-Request-Id", "D4E5:5794:51208:CD5F3:5B7D260C"}], status_code: 400}}
```
Running the same query with HTTPoison directly succeeds:
```
    url = "https://api.github.com/graphql"
    headers = [Authorization: "bearer <REDACTED>"]
    query = "{ viewer { login }}"
    query_json = "{ \"query\": \"query #{query}\" }"
    resp = HTTPoison.post(url, query_json, headers)
    IO.puts inspect resp
```
and the response:
```
{:ok, %HTTPoison.Response{body: "{\"data\":{\"viewer\":{\"login\":\"davefp\"}}}", headers: [{"Server", "GitHub.com"}, {"Date", "Wed, 22 Aug 2018 09:05:20 GMT"}, {"Content-Type", "application/json; charset=utf-8"}, {"Content-Length", "38"}, {"Status", "200 OK"}, {"X-RateLimit-Limit", "5000"}, {"X-RateLimit-Remaining", "4997"}, {"X-RateLimit-Reset", "1534931646"}, {"Cache-Control", "no-cache"}, {"X-OAuth-Scopes", ""}, {"X-Accepted-OAuth-Scopes", "repo"}, {"X-OAuth-Client-Id", "REDACTED"}, {"X-GitHub-Media-Type", "github.v4; format=json"}, {"Access-Control-Expose-Headers", "ETag, Link, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval"}, {"Access-Control-Allow-Origin", "*"}, {"Strict-Transport-Security", "max-age=31536000; includeSubdomains; preload"}, {"X-Frame-Options", "deny"}, {"X-Content-Type-Options", "nosniff"}, {"X-XSS-Protection", "1; mode=block"}, {"Referrer-Policy", "origin-when-cross-origin, strict-origin-when-cross-origin"}, {"Content-Security-Policy", "default-src 'none'"}, {"X-Runtime-rack", "0.040132"}, {"X-GitHub-Request-Id", "D4FC:5796:D5F35:1A40E9:5B7D2750"}], request_url: "https://api.github.com/graphql", status_code: 200}}
```
Any suggestions on how I can debug this would be appreciated. Thanks!

**Comment by uesteibar** [comment]
From my findings, it seems Github only accepts a json body, so I'll follow the same approach as [graphql.js](https://github.com/f/graphql.js#change-post-method) and add a `as_json` configuration to allow this behaviour.

**Comment by uesteibar** [comment]
Great catch @davefp!!

I opened [a pull request](https://github.com/uesteibar/neuron/pull/20) with this change. Could you give it a try?

You need to add the neuron dependency as 
```elixir
{:neuron, git: "https://github.com/uesteibar/neuron.git", branch: "19-queries-as-json"}
```

And add
```elixir
Neuron.Config.set(as_json: true)
```

to your app.

Please let me know how it goes! :)

**Comment by davefp** [comment]
@uesteibar That works wonderfully, thanks!

**Comment by uesteibar** [comment]
@davefp I've released it in version [v0.8.0](https://github.com/uesteibar/neuron/tree/v0.8.0). Thanks again for reporting! 🎉
